CONTRIBUTING.md

```diff
@@ -0,0 +1,106 @@
+# Contributing
+
+Pull requests are welcome!
+
+## Getting started
+
+First [fork this repo](https://help.github.com/articles/fork-a-repo/), and
+then clone it & set the upstream repo:
+
+```bash
+git clone git@github.com:your-github-username/halyard.git
+
+   # in my case (username is lwander), the above command looks like:
+   # git clone git@github.com:lwander/halyard.git
+
+cd halyard
+
+git remote add-url upstream git@github.com:spinnaker/halyard.git
+
+  # the above command allows you to push patches to the upstream repo for 
+  # approval
+```
+
+## Configuring IntelliJ
+
+In the project's root directory run
+
+```bash
+./gradlew idea
+```
+
+and then point IntelliJ at the generated project. You shouldn't need to import
+this project, simply navigating to __File__ > __Open__ > _the project's root
+directory_ should be enough.
+
+## Running the daemon
+
+In the project's root directory run
+
+```bash
+./gradlew
+```
+
+and the daemon will build & run. The first time you do this it will take a few
+minutes as it fetches the dependencies.
+
+## Running the CLI
```

Comment:
Good idea, that cli/daemon and gui/daemon interface isn't well documented outside of the rest endpoints